### PR TITLE
Eclipse Cheat Sheet

### PR DESCRIPTION
**Pull request by namandistro**
What does your Instant Answer do?

Show all The Eclipse shortcut keys as cheat sheet.

What is the data source for your Instant Answer? (Provide a link if possible)

http://www.shortcutworld.com/en/win/Eclipse.html

Why did you choose this data source?

One of the Best and even recommended.

Are there any other alternative (better) data sources?

There are chances.

Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

Coder, Eclipse Users, Git users, Geeks etc.

Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

Yes, https://duck.co/ideas/idea/6042/keyboard-shortcuts-cheatsheet-for-eclipse-ide

Are you having any problems? Do you need our help with anything?

No

What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

screenshot from 2015-08-29 12 07 23

### CODE REVIEW

**Comment by moollaza** [comment]
@namandistro looks like there's already an Eclipse Cheat Sheet under way here https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1271

I've pinged the developer as we're waiting for him to implement fixes. I'd like to give him a chance to respond and finish as he started on this first, but it was a little while ago..

File: share/goodie/cheat_sheets/eclipse.json [hunk 1]
```diff
@@ -0,0 +1,465 @@
+{
+    "id":"eclipse_cheat_sheet",
+    "name":"Eclipse",
+    "description":"Eclipse Keyboard Shortcuts ",
+    "template_type":"keyboard",
+    "metadata":{
+        "sourceName":"shortcutworld",
+        "sourceUrl":"http://www.shortcutworld.com/en/win/Eclipse.html"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/eclipse.json]: Would there be a more "official" source like http://help.eclipse.org/mars/index.jsp? How do we know if we can trust this?

File: share/goodie/cheat_sheets/eclipse.json [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+                "key":"[Ctrl] [M]",
+                "val":"Maximize or un-maximize current Editor Window (also works for other Windows)"
+            },
+            {
+                "key":"[Ctrl] [E]",
+                "val":"Show list of open Editors. Use arrow keys and enter to switch"
+            },
+            {
+                "key":"[Ctrl] [F6]",
+                "val":"Show list of open Editors. Similar to ctrl+e but switches immediately upon release of ctrl"
```
Comment (jagtalon) [inline share/goodie/cheat_sheets/eclipse.json]: Could you style "ctrl+e" as "Ctrl + e"?

Comment (jagtalon) [inline share/goodie/cheat_sheets/eclipse.json]: Same goes for the others. :)

**Comment by moollaza** [comment]
@namandistro it looks like the other PR has been updated. Would you like to submit updates to it in a new PR? Or would you like to collaborate on that PR with the other developer and get it all merged once?

I'm happy either way.

**Comment by namandistro** [comment]
@jagtalon Just made the recommend changes!!!!!
@moollaza anything will do for me.... I have no issues

**Comment by jagtalon** [comment]
@namandistro what about https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1453/files#r38263830 and https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1453/files#r38263836?

**Comment by namandistro** [comment]
@jagtalon most of our cheatsheet coming from that source 2nd the source you gave dnt really have list of anything. Talking about the small alpahabet I made every alphabet small just have a look on that

**Comment by moollaza** [comment]
@namandistro alright the Eclipse PR has been merged. You're welcome to take a look at it and add in anything you'd like :+1:

**Comment by moollaza** [comment]
@namandistro Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by namandistro** [comment]
@moollaza i had made all the changes what was needed just have a look and you commented that it will be merged!!!

**Comment by moollaza** [comment]
@namandistro sorry about the confusion. Right now this PR isn't mergeable as it's adding a new `eclipse.json` file to the `/cheat_sheets` dir. We already merged the previous `eclipse.json` file and have since moved all the `*.json` files to a new `/json` subdirectory (to separate them from the other files used by the CheatSheets Goodie.

I need you to pull master into your branch and update your PR so that it adds new content to the existing Eclipse.json file.

Does that make sense? Sorry about the hassle!

**Comment by moollaza** [comment]
@namandistro Ping! I'm just checking in to see if you're still interested in finishing this one up? We'd love to see this go live!

There's absolutely no rush, just wanted to make sure you're not stuck on anything.

**Comment by namandistro** [comment]
@moollaza can you please elaborate the issue will solve it asap....

**Comment by moollaza** [comment]
@namandistro your cheat sheet is being added to the wrong directory. It should be one level deeper in `/json/`

Right now your PR is adding a whole new file, instead of modifying the existing Cheat Sheet

**Pull request closed without merging**